src/app/services/hero/hero.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+
+import { HeroModel } from 'models/hero/hero.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HeroService {
+  create(fly: boolean, life: number): HeroModel {
+    const heroBase = this.createBaseHero({ fly, life });
+    const newHero = heroBase;
+
+    return newHero;
+  }
+
+  createHurtedHero(fly: boolean, life: number): HeroModel {
+    const heroBase = this.createBaseHero({ fly, life });
```

Comment:
const life = 20;
const heroBase = this.createBaseHero({ fly, life, name });

Reply:
shouldn't it be :
life = 20;
life, the variable pass as arg?

Reply:
You can set from param, is much better and clean :)